### PR TITLE
PLAT-887 | Allow taxonomy term deletion

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-887

### CODE REVIEW

File: resources/assets/js/admin/store/modules/taxonomyTerms.js [hunk 1]
```diff
@@ -166,6 +170,20 @@ const actions = {
 		}
 	},
 
+	async delete({commit, state}, taxonomyTerm) {
+		commit(types.SET_TAXONOMY_TERMS_SAVING, true);
+		try {
+			await axios.delete(getApiUrl(`taxonomy_terms/${taxonomyTerm.id}`));
+
+			commit(types.DELETE_TERM, taxonomyTerm);
+			commit(types.REORDER_TERMS, state.terms);
```
Comment (bognix) [inline resources/assets/js/admin/store/modules/taxonomyTerms.js]: you don't need to reorder all terms, just siblings. I highly recommend using the getChildrenByParentId getter.

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<template>
+	<div
+		v-if="term.id"
+		class="center"
+	>
+		<p class="margin bottom">
+			Czy na pewno chcesz usunąć pojęcie <em><strong>{{term.tag.name}}</strong></em> i wszystkich potomków?
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue]:
```suggestion
			Czy na pewno chcesz usunąć pojęcie <em><strong>{{term.tag.name}}</strong></em> wraz z potomkami?
```

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		</p>
+		<button
+			class="button is-primary is-danger"
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue]:
```suggestion
			class="button is-danger"
```

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	<div v-else class="notification is-info">
+		<span class="icon">
+			<i class="fa fa-info-circle"></i>
+		</span>
+		Najpierw wybierz pojęcia
+	</div>
+</template>
+
+<style lang="sass" rel="stylesheet/sass" scoped>
+	.center
```
[the remark this answers is not in the document]
Comment (kvas-damian) [reply resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue]: `.has-text-centered`

File: resources/assets/js/admin/store/modules/taxonomyTerms.js [hunk 5]
```diff
@@ -64,6 +64,10 @@ const mutations = {
 	[types.UPDATE_TERM] (state, payload) {
 		set(state.terms, state.terms.findIndex(term => term.id === payload.id), payload);
 	},
+	[types.DELETE_TERM] (state, payload) {
+		const index = state.terms.findIndex((term) => term.id === payload.id);
```
Comment (kvas-damian) [inline resources/assets/js/admin/store/modules/taxonomyTerms.js]:
```suggestion
		const index = state.terms.findIndex(term => term.id === payload.id);
```

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+			}
+
+			// TODO figure out multiple terms selected
+			return this.termById(this.selectedTerms[0]);
+		}
+	},
+	methods: {
+		...mapActions(['addAutoDismissableAlert']),
+		...mapActions('taxonomyTerms', {
+			'deleteTerm': 'delete',
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorDelete.vue]: no need to quote attribute name

**Pull request merged**